File: lab-1/domain/song.py
from domain.media_item import media_item


class Song(media_item):
    def __init__(self, title, creation_date, duration, artist):
        super().__init__(title, creation_date)
        self.duration = duration
        self.artist = artist

    def play(self):
        print(f"Playing the song: {self.title} by {self.artist}")

    def display_info(self):
        print(f"Song: {self.title} by {self.artist}")
        print(f"Duration: {self.duration} minutes")
        print(f"Release date: {self.creation_date}")

File: src/main/kotlin/media_item/implementations/Song.kt
package media_item.implementations
import media_item.MediaItem
import media_item_implementation.MediaItemImplementation
import media_item_implementation.implementations.SongImplementation


class Song(
        override val title: String,
        override val creationDate: String,
        val duration: Float,
        val artist: String,
        override val implementation: SongImplementation // Directly initialize the property here
) : MediaItem() {

    override fun displayInfo() {
        println("Song: $title by $artist")
        println("Duration: $duration minutes")
        println("Release date: $creationDate")
    }

    override fun toString(): String {
        return "Song(title='$title', creationDate='$creationDate', duration=$duration, artist='$artist')"
    }
}